rfd/0196-introducing-a-component-system.md

```diff
@@ -0,0 +1,518 @@
+the new design system. This can help reduce bugs by preventing common mistakes, and also help with maintaining a
+more consistent code style.
+
+```
+# current directory structure
+
+web/
+  packages/
+    design/
+    teleport/
+    teleterm/
+```
+
+Possible new directory structure
+
+```
+web/
+  packages/
+    legacy/ # current ESLint rules, not very strict
+      design/
+        src/
+      teleport/
+        src/
+      teleterm/
+        src/
+    design/
+      src/ # strict ESLint, TypeScript rules
+    teleport/
+      src/ # strict ESLint, TypeScript rules
+    teleterm/
+      src/ # strict ESLint, TypeScript rules
+```
+
+The "new" codebase should be able to import code from the "legacy" codebase, but not the other way around.
+
+Once we have a large portion of codebase using the new design system, and everyone is comfortable with it,
+we could then work on migrating the legacy code over.
+
+### Introduce new pages with the new design system
+
+In an ideal world, putting together new pages should be easy - taking a few reusable, common components and wiring them up.
+
+Working with the full stack team, we can gather what they components/design items they commonly reuse, and what they create each time, and start
+building out the new design system with these components in mind.
+
+### Rough timeline
+
+Implementing this change could look like this:
+
+- Adopt Chakra UI and adapt it to our theming
+- Move all the previous code into the legacy folder, across all major version branches that are currently supported
+  - This will make backporting bug fixes and changes to the previous codebase easily
+- Create new pages using the new design system
+- Start migrating the legacy code over to the new design system
+  - Some developer time will be needed for this, but would be a great hackathon idea/group effort
+- Repeat the above until legacy is no more
+
+Ideally, new pages should be indistinguishable from the old ones.
+
+## Alternatives
+
+### Ark UI
+
+Ark UI is a library that is also by the creators of Chakra UI, which provides a library of un-styled components and their functionality.
+
+As Chakra UI is built on top of this, it would be a lot of work to implement the same styling capabilities that Chakra UI
+offers, just to get to the same point.
+
+### MUI
+
+MUI is a very popular component system, with a lot of components and a large community. However, it is a lot more opinionated
+than Chakra UI - all components follow the complete Material UI design system, which would require more work to
+get them to match the Teleport design system.
+
+## Chakra usage in Teleport Identity Security
+
+Chakra UI is the component system used in Teleport Identity Security, and has been a great productivity booster
+due to the pre-built components and easy styling capabilities. Using Chakra in the Teleport codebase would provide
+a consistent design system that can be used by all Teleport products.
```

Comment:
One thing that's always troubled me about Identity Security is that the UI looks like it was built by a completely different company - it doesn't fit in with the existing Teleport UI at all.

Is this something we plan to fix, or do we get to a _consistent design system_ by throwing away the Teleport design system that we've worked so hard on the past 2-3 years and making everything look like Identity Security?

Reply:
The other way around - if we introduce Chakra here, and style it to Teleport's design system, it can then be adopted into Identity Security